Site Editor: Add page details when viewing a specific page.

scruffian opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
This adds the URL of the current page when previewing a page in the Site Editor.

Fixes https://github.com/WordPress/gutenberg/issues/48482#issuecomment-1450475528

## Why?
This gives additional context when editing a page and also makes it possible to preview the page.

## How?
Get the URL from the post data.

## Testing Instructions
1. In the Site Editor, navigate to a page using the Navigation section
2. Check that you see the page URL under the page name

## Screenshots or screencast <!-- if applicable -->
<img width="984" alt="Screenshot 2023-03-01 at 17 37 13" src="https://user-images.githubusercontent.com/275961/222218355-96df1ce1-2847-4ba7-ae18-7e4981bf3152.png">

scruffian committed:
Show the post link

scruffian committed:
Open in new window

scruffian commented on packages/edit-site/src/components/sidebar-navigation-screen-navigation-item/index.js:
```diff
@@ -49,7 +49,14 @@ export default function SidebarNavigationScreenNavigationItem() {
 			content={
-				post ? decodeEntities( post?.description?.rendered ) : null
+				<>
+					<a href={ post?.link } target="_blank" rel="noreferrer">
```
Is there a component we can use instead of a normal link?

We need to change the style of this.

jeryj replied:
It may be overkill, but we could use the `<Button />` component. It looks like if you pass it a `href` it will output it as an `<a>` element. The Preview link in the page editor uses the `<PostPreviewButton />` which wraps the wp components `<Button />` which [outputs an `<a>` element](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/button/index.tsx#L200-L206) if it has an href.

So, I think directly using an `<a>` is fine. If we want to style it (and the `<Button />` has a default that works for us, add tooltips, aria elements, and state, etc, then the `<Button />` component could be a good option.

jeryj replied:
a11y-wise it's best not to open links in a new tab without warning:

> It is recommended that when links are opened to a new window, there is advance warning.
> https://www.w3.org/TR/WCAG20-TECHS/G200.html

Using the `<Button />` element doesn't look like it automatically handles adding a visually-hidden warning if you pass an href and a `target="_blank"`, but there may be another component that does. I'm looking.

jeryj replied:
Ah, right. The the component @richtabor mentioned in the issue: `ExternalLink` lol. I'll update the PR.

jeryj committed:
Use ExternalLink component for link

For improved a11y without extra effort, let's use the ExternalLink component as it already meets WCAG specs/best practices for warning a user when a link will be opened in a new page. https://www.w3.org/TR/WCAG20-TECHS/G200.html

jeryj committed:
Style link to use gray color

Switched from the blue to a gray link. Added hover and focus styles as well. Checked color contrast and it meets WCAG AA level.

scruffian commented on packages/edit-site/src/components/sidebar-navigation-screen/style.scss:
```diff
@@ -10,6 +10,19 @@
+	}
+
+	.components-external-link__icon {
+		margin-left: $grid-unit-5;
```
This won't build for me, I get:
```
'Error: Undefined variable.\n' +
    '   ╷\n' +
    '22 │         margin-left: $grid-unit-5;\n' +
    '   │                      ^^^^^^^^^^^^\n' +
    '   ╵\n' +
```

scruffian replied:
Pushed a fix for this

scruffian replied:
I'm surprised this isn't part of the component itself. It might be worth looking into a more general fix for the component, in a followup PR...

richtabor replied:
> I'm surprised this isn't part of the component itself. It might be worth looking into a more general fix for the component, in a followup PR...

I think it's cause this can be used in smaller inline text, where its not necessary to add space before the icon. Although we probably should have some space after the icon in the screenshot: 

<img width="279" alt="CleanShot 2023-03-01 at 17 47 22" src="https://user-images.githubusercontent.com/1813435/222282979-3a2ce47d-f221-4981-8b00-4a754594bb41.png">

richtabor replied:
> Pushed a fix for this

Did you push it?

scruffian replied:
oops no, I have now!

jeryj replied:
Weird — it both built and showed a space as if it was working for me 🤷🏻 Thanks for catching and fixing that!

scruffian commented on packages/edit-site/src/components/sidebar-navigation-screen-navigation-item/index.js:
```diff
@@ -49,7 +52,19 @@ export default function SidebarNavigationScreenNavigationItem() {
+					{ post?.link ? (
+						<ExternalLink
+							className="edit-site-sidebar-navigation-screen__page-link"
+							href={ post.link }
```
Should this be `post?.link`, in case the post isn't defined?

scruffian replied:
Oh wait I see we already check that earlier.

scruffian commented on packages/edit-site/src/components/sidebar-navigation-screen-navigation-item/index.js:
```diff
@@ -49,7 +52,19 @@ export default function SidebarNavigationScreenNavigationItem() {
+						<ExternalLink
+							className="edit-site-sidebar-navigation-screen__page-link"
+							href={ post.link }
+						>
```
Same question

scruffian commented:
Thanks for the fixes @jeryj, I'm happy for this to be merged if you are :)

scruffian committed:
fix the grid unit variable name

jameskoster reviewed (approved):
<img width="359" alt="Screenshot 2023-03-02 at 10 19 33" src="https://user-images.githubusercontent.com/846565/222400667-3f7839e9-d506-499d-9e44-53c6a6ad375d.png">

We still need to fix the alignment, and may want to follow-up to add more information (publish date, that kind of thing). But this seems like a fair starting point and solves the chasm of emptiness we have currently.

ntsekouras commented:
I just cherry-picked this PR to the wp/6.2 branch to get it included in the next release: 0641bceecf
